expression/typeinfer_test.go

```diff
@@ -2965,9 +2965,8 @@ func (s *InferTypeSuite) createTestCase4Literals() []typeInferTestCase {
 
 func (s *InferTypeSuite) createTestCase4JSONFuncs() []typeInferTestCase {
 	return []typeInferTestCase{
-		{"json_type(c_json)", mysql.TypeVarString, charset.CharsetUTF8MB4, 0, 51, types.UnspecifiedLength},
-		// TODO: flen of json_unquote doesn't follow MySQL now.
-		{"json_unquote(c_json)", mysql.TypeVarString, charset.CharsetUTF8MB4, 0, mysql.MaxFieldVarCharLength, types.UnspecifiedLength},
+		{"json_type(c_json)", mysql.TypeVarString, charset.CharsetUTF8MB4, mysql.BinaryFlag, 51, types.UnspecifiedLength},
+		{"json_unquote(c_json)", mysql.TypeLongBlob, charset.CharsetUTF8MB4, mysql.BinaryFlag, mysql.MaxLongBlobWidth, types.UnspecifiedLength},
```

Comment:
Quote and unquote seems to have a decimals of 31 on mysql8.

```
Type:       LONG_BLOB
Collation:  utf8mb4_general_ci (45)
Length:     4294967295
Max_length: 0
Decimals:   31
Flags:      BINARY
```

Reply:
it MySQL, 31 is `DECIMAL_NOT_SPECIFIED` see https://github.com/mysql/mysql-server/blob/fbdaa4def30d269bc4de5b85de61de34b11c0afc/include/decimal.h#L152-L153